src/privacy.rs

```diff
@@ -0,0 +1,276 @@
+    let (should_reload, path) = {
+        let global_context_lock = global_context.read().await;
+        let current_time = SystemTime::now().duration_since(SystemTime::UNIX_EPOCH).unwrap().as_secs();
+        let should_reload = global_context_lock.privacy_settings.expiry_time <= current_time;
+        let path = global_context_lock.cache_dir.join("privacy.yaml");
+        (should_reload, path)
+    };
+
+    if !should_reload {
+        return;
+    }
+
+    if !path.exists() {
+        match std::fs::File::create(&path) {
+            Ok(mut file) => {
+                if let Err(e) = file.write_all(COMPILED_IN_INITIAL_PRIVACY_YAML.as_bytes()) {
+                    error!("Failed to write to file: {}", e);
+                }
+            }
+            Err(e) => {
+                error!("Failed to create file: {}", e);
+            }
+        }
+    }
+
+    let new_privacy_settings = read_privacy_yaml(&path).await;
+
+    let mut global_context_lock = global_context.write().await;
+    global_context_lock.privacy_settings = Arc::new(new_privacy_settings);
+}
+
+fn any_glob_matches_path(globs: &Vec<String>, path: &Path) -> bool {
+    globs.iter().any(|glob| {
+        let pattern = Pattern::new(glob).unwrap();
+        pattern.matches_path(path)
+    })
+}
+
+async fn get_file_privacy_level(global_context: Arc<ARwLock<GlobalContext>>, path: &Path) -> FilePrivacyLevel {
+    let global_context_lock = global_context.read().await;
+    let privacy_settings = &global_context_lock.privacy_settings;
+
+    if any_glob_matches_path(&privacy_settings.file_privacy.blocked, path) {
+        FilePrivacyLevel::Blocked
+    } else if any_glob_matches_path(&privacy_settings.file_privacy.only_send_to_servers_I_control, path) {
+        FilePrivacyLevel::OnlySendToServersIControl
+    } else {
+        FilePrivacyLevel::AllowToSendEverywhere
+    }
+}
+
+pub async fn check_file_privacy(global_context: Arc<ARwLock<GlobalContext>>, path: &Path, max_allowed_privacy_level: &FilePrivacyLevel) -> Result<(), String> {
+    load_privacy_if_needed(global_context.clone()).await;
+
+    let file_privacy_level = get_file_privacy_level(global_context.clone(), path).await;
+    if file_privacy_level > *max_allowed_privacy_level {
+        Err(format!("File privacy level for file is too restrictive, {} is {:?}", path.display(), file_privacy_level))
+    } else {
+        Ok(())
+    }
+}
+
+pub fn check_file_privacy_sync(global_context: Arc<ARwLock<GlobalContext>>, path: &Path, max_allowed_privacy_level: &FilePrivacyLevel) -> Result<(), String> {
+    let file_privacy_level = futures::executor::block_on(async {
+        load_privacy_if_needed(global_context.clone()).await;
+        get_file_privacy_level(global_context.clone(), path).await
+    });
+
+    if file_privacy_level > *max_allowed_privacy_level {
+        Err(format!("File privacy level for file is too restrictive, {} is {:?}", path.display(), file_privacy_level))
+    } else {
+        Ok(())
+    }
+}
```

Comment:
bad idea (ask chat why)

Reply:
chat tells me to remove the else statement to avoid innecesary flow control
```
    if file_privacy_level > *max_allowed_privacy_level {
        Err(format!("File privacy level for file is too restrictive, {} is {:?}", path.display(), file_privacy_level))
    } 
    
    Ok(())
    ``` is it what you meant?